src/cpp/fastdds/xtypes/dynamic_types/TypeValueConverter.hpp

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FASTDDS_XTYPES_DYNAMIC_TYPES_TYPEVALUECONVERTER_HPP
+#define FASTDDS_XTYPES_DYNAMIC_TYPES_TYPEVALUECONVERTER_HPP
+
+#include "TypeForKind.hpp"
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+namespace detail {
+
+struct converter
+{
+    const std::string& x;
+    template<typename T> operator T()
+    {
+        return 0;
+    }
+
+};
+
+
+template <> inline converter::operator TypeForKind<TK_UINT8>()
+{
+    return static_cast<uint8_t>(std::stoul(x));
+}
+
+template <> inline converter::operator TypeForKind<TK_INT16>()
+{
+    return static_cast<int16_t>(std::stoi(x));
```

Comment:
Done in https://github.com/eProsima/Fast-DDS/commit/df7e363e81e00d106839a54162b2cc8032faec18

Reply:
This suggestion has not been applied correctly. `std::stoi` converts the string into a int value, whereas the `std::stol` converts into a long value. That is the reason why I suggested removing the casting but leaving the stoi method. Although it is true that the C++ specification does not force a specific type size, so it could be safer to keep the casting. In that case, I think the casting should be done in other cases as `TK_INT32`

Reply:
I'm using `std::stol` to be similar as in TK_UINT16 case which uses `std::stoul`.